### ISSUE TITLE
Docs Question

### ISSUE THREAD

**Issue by evanbechtol** [scandum/quadsort#3]
First of all, great job on this and I am currently working on trying to make a JavaScript port  and test out against various sorting algorithms on Node.js.

I don't quite understand these two sections, or the symbols. Can you explain in more detail?
```
   A --> A,B --> A,C
   B --> A,B --> A,B,C,D
   C --> C,D --> A,B,C,D
   D --> C,D --> B,D
```

```
   A --> A,B --> A? --> C? --> A,C
   B --> A,B --> B? --> D? --> A,C
   C --> C,D --> C? --> A? --> B,D
   D --> C,D --> D? --> B? --> B,D
```

Side question, why are the variable names so obscure? Variable names go a long way to improve readability, and it's really **really** hard to follow the flow of this code.

**Comment by scandum** [comment]
I updated the diagram and hopefully it's a little clearer now.

pta = pointer to array
pts = pointer to swap

a and b are typically used as the variable name in merge sorts. I ended up picking c and d on a whim.

**Comment by evanbechtol** [comment]
Thanks for the follow-up! I believe that I have the `quad_swap` function working in JS. Starting on the meat of the algorithm now. 

Since I am not proficient in C, I don't have to worry about the memory stuff (fortunately for me). ;)

What are [these variables](https://github.com/scandum/quadsort/blob/0add1c1c758d5b39023350559539496d415db2d9/quadsort.c#L271) for?
```
*c, *c_max, *d, *d_max, *end
```

**Comment by uahnbu** [comment]
@evanbechtol Have you made any progress in porting quadsort to js?
